CodeMaid/Logic/Cleaning/UsingStatementCleanupLogic.cs

```diff
@@ -91,20 +85,25 @@ public void RemoveAndSortUsingStatements(TextDocument textDocument)
             else
             {
                 _commandHelper.ExecuteCommand(textDocument, "Edit.RemoveUnusedUsings");
-                _commandHelper.ExecuteCommand(textDocument, "Edit.SortUsings");
             }
 
-            // Check each using statement point and re-insert it if removed.
-            foreach (var point in points)
+            // Ignore any using statements that are still referenced
+            usingStatementsToReinsert = usingStatementsToReinsert
+                 .Where(usingStatement => TextDocumentHelper.FirstOrDefaultMatch(textDocument, string.Format(patternFormat, usingStatement)) == null)
+                 .ToList();
+
+            ThreadHelper.ThrowIfNotOnUIThread();
+            var point = textDocument.StartPoint.CreateEditPoint();
+
+            foreach (string usingStatement in usingStatementsToReinsert)
             {
-                string text = point.editPoint.GetLine();
-                if (text != point.text)
-                {
-                    point.editPoint.StartOfLine();
-                    point.editPoint.Insert(point.text);
-                    point.editPoint.Insert(Environment.NewLine);
-                }
+                point.StartOfLine();
+                point.Insert(usingStatement);
+                point.Insert(Environment.NewLine);
             }
+
+            // Now sort without removing to ensure correct ordering.
+            _commandHelper.ExecuteCommand(textDocument, "Edit.SortUsings");
```

Comment:
This will cause the usings to be sorted twice for newer versions of the IDE, even for the default case where the using statements to re-insert do not exist.  I can see why it is split out to happen after re-insertion, but I think it should be guarded to avoid extra work.

Reply:
Good catch. I'll update code to skip that work if there are no reinsert candidates.